actions/admin/utxos.go

```diff
@@ -16,75 +18,44 @@ import (
+// @Success		200 {object} []response.Utxo "List of utxos"
 // @Failure		400	"Bad request - Error while parsing SearchUtxos from request body"
 // @Failure 	500	"Internal server error - Error while searching for utxos"
-// @Router		/v1/admin/utxos/search [post]
+// @Router		/api/v1/admin/utxos [get]
 // @Security	x-auth-xpub
 func utxosSearch(c *gin.Context, _ *reqctx.AdminContext) {
 	logger := reqctx.Logger(c)
-	var reqParams filter.AdminSearchUtxos
-	if err := c.Bind(&reqParams); err != nil {
-		spverrors.ErrorResponse(c, spverrors.ErrCannotBindRequest, logger)
+
+	searchParams, err := query.ParseSearchParams[filter.AdminUtxoFilter](c)
+	if err != nil {
+		spverrors.ErrorResponse(c, spverrors.ErrCannotParseQueryParams, logger)
 		return
 	}
 
-	conditions, err := reqParams.Conditions.ToDbConditions()
+	conditions, err := searchParams.Conditions.ToDbConditions()
 	if err != nil {
-		spverrors.ErrorResponse(c, spverrors.ErrInvalidConditions, logger)
+		spverrors.ErrorResponse(c, spverrors.ErrInvalidConditions.WithTrace(err), logger)
 		return
 	}
+	metadata := mappings.MapToMetadata(searchParams.Metadata)
+	pageOptions := mappings.MapToDbQueryParams(&searchParams.Page)
 
 	utxos, err := reqctx.Engine(c).GetUtxos(
 		c.Request.Context(),
-		mappings.MapToMetadata(reqParams.Metadata),
+		metadata,
 		conditions,
-		mappings.MapToQueryParams(reqParams.QueryParams),
+		pageOptions,
 	)
 	if err != nil {
-		spverrors.ErrorResponse(c, err, logger)
-		return
-	}
-
-	c.JSON(http.StatusOK, utxos)
-}
-
-// utxosCount will count all utxos filtered by metadata
-// Count utxos filtering by metadata godoc
-// @Summary		Count utxos
-// @Description	Count utxos
-// @Tags		Admin
-// @Produce		json
-// @Param		CountUtxos body filter.AdminCountUtxos false "Enables filtering of elements to be counted"
-// @Success		200	{number} int64 "Count of utxos"
-// @Failure		400	"Bad request - Error while parsing CountUtxos from request body"
-// @Failure 	500	"Internal Server Error - Error while fetching count of utxos"
-// @Router		/v1/admin/utxos/count [post]
-// @Security	x-auth-xpub
-func utxosCount(c *gin.Context, _ *reqctx.AdminContext) {
-	logger := reqctx.Logger(c)
-	var reqParams filter.AdminCountUtxos
-	if err := c.Bind(&reqParams); err != nil {
-		spverrors.ErrorResponse(c, spverrors.ErrCannotBindRequest, logger)
-		return
-	}
-
-	conditions, err := reqParams.Conditions.ToDbConditions()
-	if err != nil {
-		spverrors.ErrorResponse(c, spverrors.ErrInvalidConditions, logger)
+		spverrors.ErrorResponse(c, spverrors.ErrCouldNotFindUtxo.WithTrace(err), logger)
 		return
 	}
 
-	count, err := reqctx.Engine(c).GetUtxosCount(
-		c.Request.Context(),
-		mappings.MapToMetadata(reqParams.Metadata),
-		conditions,
-	)
-	if err != nil {
-		spverrors.ErrorResponse(c, err, logger)
-		return
+	contracts := make([]*response.Utxo, 0)
```

Comment:
I know that it is moved from the old code.
However, could you reserve the memory to slightly boost it up?
```suggestion
	contracts := make([]*response.Utxo, 0, len(utxos))
```

Could you also, search for similar cases in your changes and do the same?

Reply:
good you pointing this as i was wondering about, but i left this way as most ppl was following this pattern